ibis/expr/rules.py

```diff
@@ -424,11 +424,17 @@ def table(arg, *, schema=None, **kwargs):
 
 @validator
 def column_from(name, column, *, this):
-    if not isinstance(column, (str, int)):
-        raise com.IbisTypeError(
-            f"value must be an int or str, got {type(column).__name__}"
-        )
-    return getattr(this, name)[column]
+    if isinstance(column, ir.AnyColumn) and hasattr(column.op(), "name"):
```

Comment:
It might also be useful to short circuit here and raise an error if the column doesn't have a name:

```python
if isinstance(column, ir.ColumnExpr):
    col_name = column.get_name()  # this will raise an error if the column isn't named
    if col_name != column:
        # raise a useful error message
    # now check for equality
```

Reply:
This should still be a `com.IbisTypeError`?

Reply:
Yep.